.github/workflows/ci.yaml

```diff
@@ -18,6 +18,18 @@ jobs:
           path: ~/.cache/pip
           key: ${{ hashFiles('pyproject.toml') }}
       - uses: pre-commit/action@v2.0.3
+
+  Type-Check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+          architecture: 'x64'
+      - run: pip3 install nox
+      - run: nox -s type_check
```

Comment:
Can we adding caching like the one we have for tests here too please, so the CI doesn't need to install everything again.

Reply:
Same as

https://github.com/astro-projects/astro/blob/c57209b8c8b0c28f0206b09d4c01eec6a69103db/.github/workflows/ci.yaml#L116-L121

Reply:
added suggestion here: https://github.com/astro-projects/astro/pull/296/files#r844991016